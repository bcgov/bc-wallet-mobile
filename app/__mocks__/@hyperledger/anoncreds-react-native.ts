const AnonCredsReactNative = {
  install: jest.fn(),
}

export default AnonCredsReactNative
