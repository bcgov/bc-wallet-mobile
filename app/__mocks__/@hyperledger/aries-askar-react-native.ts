const AriesAskarReactNative = {
  install: jest.fn(),
}

export default AriesAskarReactNative
