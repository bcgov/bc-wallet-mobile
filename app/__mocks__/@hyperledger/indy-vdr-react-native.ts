const IndyVDRReactNative = {
  install: jest.fn(),
}

export default IndyVDRReactNative
