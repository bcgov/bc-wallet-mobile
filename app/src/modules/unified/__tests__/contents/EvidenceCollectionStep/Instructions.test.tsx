import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import InstructionsContent from '../../../contents/EvidenceCollectionStep/Instructions'
import { testIdWithKey } from '@hyperledger/aries-bifold-core'

describe('InstructionsContent Component', () => {
  const goToScan = jest.fn()
  const goToManualSerial = jest.fn()

  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders correctly', () => {
    const tree = render(<InstructionsContent goToScan={goToScan} goToManualSerial={goToManualSerial} />)
    expect(tree).toMatchSnapshot()
  })

  test('ScanBarcode test ID is present and press handler works', () => {
    const { getByTestId } = render(<InstructionsContent goToScan={goToScan} goToManualSerial={goToManualSerial} />)

    const continueButton = getByTestId(testIdWithKey('ScanBarcode'))
    expect(continueButton).toBeDefined()
    fireEvent(continueButton, 'press')
    expect(goToScan).toHaveBeenCalledTimes(1)
  })

  test('EnterManually test ID is present and press handler works', () => {
    const { getByTestId } = render(<InstructionsContent goToScan={goToScan} goToManualSerial={goToManualSerial} />)

    const continueButton = getByTestId(testIdWithKey('EnterManually'))
    expect(continueButton).toBeDefined()
    fireEvent(continueButton, 'press')
    expect(goToManualSerial).toHaveBeenCalledTimes(1)
  })
})
