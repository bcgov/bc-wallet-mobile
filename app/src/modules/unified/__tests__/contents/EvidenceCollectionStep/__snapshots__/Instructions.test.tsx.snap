// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionsContent Component renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    {
      "backgroundColor": "#313132",
      "height": "100%",
      "justifyContent": "space-between",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flex": 1,
        "padding": 24,
      }
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          {
            "default": "",
          }
        }
        style={
          {
            "height": 470.34000000000003,
            "marginBottom": 24,
            "width": 702,
          }
        }
      />
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 26,
            "fontWeight": "bold",
            "marginBottom": 16,
          }
        }
      >
        Unified.Instructions.Heading
      </Text>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginBottom": 8,
          }
        }
      >
        Unified.Instructions.Paragraph
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": "auto",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Unified.Instructions.ScanBarcode"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/ScanBarcode"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
              false,
              false,
            ]
          }
        >
          Unified.Instructions.ScanBarcode
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 16,
        }
      }
    >
      <View
        accessibilityLabel="Unified.Instructions.EnterManually"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/EnterManually"
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#42803E",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
                false,
                false,
              ]
            }
          >
            Unified.Instructions.EnterManually
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
